ql/src/java/org/apache/hadoop/hive/ql/exec/tez/ReduceRecordProcessor.java

```diff
@@ -305,6 +305,14 @@ public void abort() {
     } else {
       LOG.info("reducer not setup yet. abort not being forwarded");
     }
+    if (reduceWork != null) {
```

Comment:
do we need to abort `mergeWorkList`?

Reply:
Here just to set dummyop's abort to true.
In my bug scenario, abort was incorrectly set from true to false. Therefore, when dummyOp.close(abort) was called, it would cause unexpected commit file. If dummyOp's abort was set to true here, it would not commit.

> Note: If `for (ReduceRecordSource rs: sources)` is handled correctly, there is no problem if it is not modified here. I just think that when abort is set to true in one place, it means that this task attempt is useless, so all ops should be set to true.

Reply:
yep